packages/donor/src/screens/bookDonation/BookDonationScreenContainer.tsx

```diff
@@ -34,22 +40,53 @@ export function BookDonationScreenContainer({
   }
 
   const onSlotSelected = (donationSlot: DonationSlotToBook) => {
-    appointmentToBookStore.setAppointmentToBook(donationSlot);
-    if (isLoggedIn) {
-      navigate(MainNavigationKeys.Questionnaire);
+    setSelectedSlot(donationSlot);
+    const currentDonor = appState.donor;
+    const timeDifference = currentDonor?.lastCompletedDonationTimeMillis
+      ? DateUtils.getNumberOfDaysBetweenDates(
+          donationSlot.donationStartTimeMillis,
+          currentDonor.lastCompletedDonationTimeMillis
+        )
+      : MIN_DAYS_BETWEEN_DONATIONS + 1;
```

Comment:
IIUC, if the donor has no last donation, you set the last donation to `MIN_DAYS_BETWEEN_DONATIONS + 1` so that the if will fail later.

Let me suggest a cleaner approach:
```
const isAppointmentTooCloseToLastDonation = currentDonor.lastCompletedDonationTimeMillis && DateUtils.getNumberOfDaysBetweenDates(
          donationSlot.donationStartTimeMillis,
          currentDonor.lastCompletedDonationTimeMillis
        ) < MIN_DAYS_BETWEEN_DONATIONS
``` 

Then use this boolean in the if below